### PR TITLE
added missing includes

### PR DESCRIPTION
**Pull request by sherpya**
stdlib for calloc/malloc etc, other windows include to build if you define WIN32_LEAN_AND_MEAN
(I've only tested wingui and wincon)

### CODE REVIEW

File: wingui/pdcdisp.c [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <tchar.h>
+#include <commdlg.h>
```
Comment (GitMensch) [inline wingui/pdcdisp.c]: What do we need this additional include for? I'm asking as this is not part of the PR issue text.

File: wingui/pdcscrn.c [hunk 2]
```diff
@@ -1,7 +1,9 @@
 /* Public Domain Curses */
 
 #include "pdcwin.h"
+#include <shellapi.h>
```
Comment (GitMensch) [inline wingui/pdcscrn.c]: What do we need this additional include for? I'm asking as this is not part of the PR issue text.

**Comment by GitMensch** [comment]
Warning: I've took care of updating the PR base fork to fix merge conflicts, allowing a possible auto-merge of this PR but did not compiled the result and am not sure about the necessary addition headers that would get included.
@sherpya Can you please recheck if these additional includes are necessary in the current state? Which warnings do they solve?

File: wingui/pdcscrn.c (hunk 2 above)
Comment (Bill-Gray) [reply wingui/pdcscrn.c]: Similar issue to above,  except this time,  `ExtractIcon()` is missing in lean/mean mode.

**Comment by Bill-Gray** [comment]
[wmcbrine set `WIN32_LEAN_AND_MEAN` in `wincon/pdcwin.h`](https://github.com/wmcbrine/PDCurses/commit/8701f07845b1a7bfdc3bc95b2402d5ab0b4c8379) about two years ago,  for "slightly faster Windows compilation".  Now that we're using his code for WinCon,  we're actually all set for that platform now.
I tried similar compilation for WinGUI, with both MSVC and cross-compiling with MinGW from Linux,  and got the same complaints about missing font dialogs and `ExtractIcon`.  So,  thank you,  @sherpya;  your proposed patch of adding the `#include` files fixed both problems.  I've added the fixes as commit c788b54.
